demos/classification_demo/python/classification_demo.py

```diff
@@ -0,0 +1,289 @@
+
+    metrics = PerformanceMetrics()
+    render_metrics = PerformanceMetrics()
+    presenter = None
+    output_transform = None
+    video_writer = cv2.VideoWriter()
+    ESC_KEY = 27
+    key = -1
+    while True:
+        if async_pipeline.callback_exceptions:
+            raise async_pipeline.callback_exceptions[0]
+        # Process all completed requests
+        results = async_pipeline.get_result(next_frame_id_to_show)
+        if results:
+            classifications, frame_meta = results
+            frame = frame_meta['frame']
+            start_time = frame_meta['start_time']
+            if args.raw_output_message:
+                print_raw_results(classifications, next_frame_id_to_show)
+
+            presenter.drawGraphs(frame)
+            rendering_start_time = perf_counter()
+            frame = draw_labels(frame, classifications, output_transform)
+            if delay or args.no_show:
+                render_metrics.update(rendering_start_time)
+                metrics.update(start_time, frame)
+
+            if video_writer.isOpened() and (args.output_limit <= 0 or next_frame_id_to_show <= args.output_limit-1):
+                video_writer.write(frame)
+            next_frame_id_to_show += 1
+
+            if not args.no_show:
+                cv2.imshow('Classification Results', frame)
+                key = cv2.waitKey(delay)
+                # Quit.
+                if key in {ord('q'), ord('Q'), ESC_KEY}:
+                    break
+                presenter.handleKey(key)
+            continue
+
+        if async_pipeline.is_ready():
+            # Get new image/frame
+            start_time = perf_counter()
+            frame = cap.read()
+            if frame is None:
+                if next_frame_id == 0:
+                    raise ValueError("Can't read an image from the input")
+                break
+            if next_frame_id == 0:
+                output_transform = models.OutputTransform(frame.shape[:2], args.output_resolution)
+                if args.output_resolution:
+                    output_resolution = output_transform.new_resolution
+                else:
+                    output_resolution = (frame.shape[1], frame.shape[0])
+                presenter = monitors.Presenter(args.utilization_monitors, 55,
+                                               (round(output_resolution[0] / 4), round(output_resolution[1] / 8)))
+                if args.output and not video_writer.open(args.output, cv2.VideoWriter_fourcc(*'MJPG'),
+                                                         cap.fps(), output_resolution):
+                    raise RuntimeError("Can't open video writer")
+            # Submit for inference
+            async_pipeline.submit_data(frame, next_frame_id, {'frame': frame, 'start_time': start_time})
+            next_frame_id += 1
+
+        else:
+            # Wait for empty request
+            async_pipeline.await_any()
+
+    if key not in {ord('q'), ord('Q'), ESC_KEY}:
```

Comment:
What's the purpose of this check? I launched the demo with a video and then exit by escape, the demo always hangs out here

Reply:
I'll recheck it

Reply:
Probably the moving of `await_all()` before the condition resolves the issue